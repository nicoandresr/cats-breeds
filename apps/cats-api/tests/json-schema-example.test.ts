import { findAndParseConfig } from '@graphql-mesh/cli';
import { getMesh } from '@graphql-mesh/runtime';
import { readFile } from 'fs-extra';
import { join } from 'path';

import { printSchema, lexicographicSortSchema } from 'graphql';

const mesh$ = findAndParseConfig({
  dir: join(__dirname, '../config'),
}).then((config) => getMesh(config));

describe('JSON Schema Example', () => {
  it('should generate correct schema', async () => {
    const { schema } = await mesh$;
    expect(
      printSchema(lexicographicSortSchema(schema))
    ).toMatchSnapshot();
  });
  it('should give correct response', async () => {
    const { execute } = await mesh$;
    const query = await readFile(
      join(__dirname, '../src/example-query.graphql'),
      'utf8'
    );
    const result = await execute(query, {});
    expect(result?.data?.breeds).toHaveLength(2);
    expect(result?.data?.breeds[0]?.id).toBeDefined();
  });
  afterAll(() => mesh$.then((mesh) => mesh.destroy()));
});
