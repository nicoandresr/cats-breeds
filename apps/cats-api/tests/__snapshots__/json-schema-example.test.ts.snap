// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JSON Schema Example should generate correct schema 1`] = `
"directive @globalOptions(endpoint: String, operationHeaders: ObjMap, queryParams: ObjMap, queryStringOptions: ObjMap, sourceName: String) on OBJECT

directive @httpOperation(httpMethod: HttpMethod, isBinary: Boolean, operationSpecificHeaders: ObjMap, path: String, queryParamArgMap: ObjMap, queryStringOptionsByParam: ObjMap, requestBaseBody: ObjMap) on FIELD_DEFINITION

type Cat {
  adaptability: Int
  affectionLevel: Int
  altNames: String
  cfaUrl: String
  childFriendly: Int
  countryCode: String
  countryCodes: String
  description: JSON
  dogFriendly: Int
  energyLevel: Int
  experimental: Int
  grooming: Int
  hairless: Boolean
  healthIssues: Int
  hypoallergenic: Boolean
  id: String
  indoor: Boolean
  intelligence: Int
  lap: Boolean
  lifeSpan: String
  name: String
  natural: Boolean
  origin: String
  rare: Int
  referenceImageId: String
  rex: Int
  sheddingLevel: Int
  shortLegs: Boolean
  socialNeeds: Int
  strangerFriendly: Int
  suppressedTail: Boolean
  temperament: String
  vcahospitalsUrl: String
  vetstreetUrl: String
  vocalisation: Int
  weight: QueryBreedsItemsWeight
  wikipediaUrl: String
}

enum HttpMethod {
  CONNECT
  DELETE
  GET
  HEAD
  OPTIONS
  PATCH
  POST
  PUT
  TRACE
}

\\"\\"\\"
The \`JSON\` scalar type represents JSON values as specified by [ECMA-404](http://www.ecma-international.org/publications/files/ECMA-ST/ECMA-404.pdf).
\\"\\"\\"
scalar JSON @specifiedBy(url: \\"http://www.ecma-international.org/publications/files/ECMA-ST/ECMA-404.pdf\\")

scalar ObjMap

type Query {
  \\"\\"\\"/breeds\\"\\"\\"
  breeds(Content_Type: String, limit: Int, page: Int): [Cat]

  \\"\\"\\"/breeds/:breed_id\\"\\"\\"
  breedsByBreedId(Content_Type: String, breed_id: String!): JSON
}

type QueryBreedsItemsWeight {
  imperial: String
  metric: String
}"
`;
